js/util/script_detection.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const verticalRegExp = new RegExp([
+    '[ᄀ-ᇿ]',
+    '[가-힣]',
+    '[ㄱ-ㆎ]',
+    '[一-鿌]',
+    '[㐀-䶵]',
+    // eslint-disable-next-line no-irregular-whitespace
+    '[　-〿]',
+    '\uD840[\uDC00-\uFFFF]|[\uD841-\uD872]|\uD873[\u0000-\uDEAF]', // '[𠀀-𬺯]'
+    '[！-￮]',
```

Comment:
Per https://github.com/mapbox/mapbox-gl-js/pull/3402#issuecomment-254670626, we should restrict this to fullwidth forms only (`[！-｠￠-￦]`) since halfwidth forms would look quite odd unless doubled up.

/cc @friedbunny